src/excmds.ts

```diff
@@ -2740,15 +2743,20 @@ export async function winopen(...args: string[]) {
         switch (args[0]) {
             case "-private":
                 createData.incognito = true
-                args = args.slice(1, args.length)
+                args.shift()
                 firefoxArgs = "--private-window"
                 break
 
             case "-popup":
                 createData.type = "popup"
-                args = args.slice(1, args.length)
+                args.shift()
                 break
 
+            case "-c":
+                if (args.length < 2) throw new Error(`You must provide a container name!`)
+                args.shift()
+                return composite(`tabopen -c ${args.join(" ")} | sleep 100 | tabdetach`)
```

Comment:
```suggestion
                return composite(`tabopen -c ${args.join(" ")}; sleep 100; tabdetach`)
```

`;` means run one after the other, `|` means pipe the results of each one into the next.

I don't like the `sleep 100` so if we can avoid it at all I'd like that.

Reply:
@bovine3dom: Are referring to #2698?

I think something along the lines of 
`
return tabopen(args).then(() =>  tabdetach)
`
should already work, but I'll give it a try.

Reply:
I think I'm referring to the problem I was fixing in that PR - that is, that `openInTab` doesn't work very well because `await [make a new tab]` returns more quickly than Tridactyl loading in that tab.

We probably need to make Tridactyl fire an event to the background script that says it has been loaded in a tab. We can then listen for that event and run `openInTab` then?

That said, `openInTab` runs in the background so it shouldn't matter ...

Give it your way a go and see what happens : )

Reply:
Sleepless in ```winopen```. It seems to work - was surprised myself.